### ISSUE TITLE
Add debug logging for protocol output in the python driver

### ISSUE THREAD

**Issue by deontologician** [rethinkdb/rethinkdb#4812]
It would be useful to be able to see the output of the driver's wire format for driver writers.

Something like:

```py
import logging
logger = logging.getLogger("rethinkdb.driver")

...
# inside the .build() method somewhere
logger.debug("Serializing {} to {}", term, built_term)
```

**Comment by danielmewes** [comment]
Also linking to https://github.com/neumino/rethinkdb-driver-development here, which basically implements this (but is somewhat outdated).

**Comment by neumino** [comment]
I didn't update this tool since 1.13. The tool was mostly to print a human readable version of the protobuf message. Now that we switched to the JSON protocol, printing the message sent to the server is easier (`console.log(query.build())` does the trick in the JS driver if I'm not mistaken)

**Comment by bchavez** [comment]
I was recently looking for a better way to debug the JSON protocol for my C# driver. I previously used @neumino 's driver tool to help debug AST terms but as mentioned, it's a bit outdated.

I did find a very decent work around though... I'm using the web-admin console to debug the JSON protocol and AST. I couldn't get @neumino 's  `console.log(query.build())` to work out of the box but I was able to disable the web-admin's refresh timers so I could easily see the network traffic being sent over the wire.

The details to disable the web-admin refresh timers are here:

https://github.com/bchavez/RethinkDb.Driver#query-debugging

Basically once the "refresh" timers are disabled, the network tab in the dev console is a lot more quiet. The `?conn_id=value` request payloads have the AST when you hit the **Run** button. Using the web-admin tool is great because you also get a nice view of the data that gets returned.

![screen511](https://cloud.githubusercontent.com/assets/478118/10414140/94b97cb0-6f79-11e5-9ca2-afc35df507f5.png)

Hope that helps,
Brian

**Comment by neumino** [comment]
@bchavez -- Sorry for being more explicit earlier. I used to print stuff in the console, but you can actually do this, which gives you a pretty nice output:

```
r.expr(
  r.table("foo").map(function(x) { return x.merge({foo: "bar"})}).build()
  // Or <your_query>.build()
)
```

**Comment by neumino** [comment]
@bchavez -- pretty cool driver too :-)

**Comment by bchavez** [comment]
Cool! Thank you @neumino . The `.build()` tip will really help. The challenge now is getting [2278 C# auto-generated unit tests](https://github.com/bchavez/RethinkDb.Driver/tree/master/Source/RethinkDb.Driver.Tests/Generated) from `rethinkdb\test\rql_test\src` to pass. :8ball: :100:
